exporter/prometheusremotewriteexporter/compliance/compliance_test.go

```diff
@@ -0,0 +1,188 @@
+`
+
+var (
+	logger = log.NewLogfmtLogger(log.NewSyncWriter(os.Stderr))
+	tests  = []func() cases.Test{
+		// Test each type.
+		cases.CounterTest,
+		cases.GaugeTest,
+		cases.HistogramTest,
+		cases.SummaryTest,
+
+		// Test Up metrics.
+		cases.UpTest,
+		cases.InvalidTest,
+
+		// Test for various labels
+		cases.JobLabelTest,
+		cases.InstanceLabelTest,
+		cases.SortedLabelsTest,
+		cases.RepeatedLabelsTest,
+		cases.EmptyLabelsTest,
+		cases.NameLabelTest,
+		cases.HonorLabelsTest,
+
+		// Other misc tests.
+		cases.StalenessTest,
+		cases.TimestampTest,
+		cases.HeadersTest,
+		cases.OrderingTest,
+		cases.Retries500Test,
+		cases.Retries400Test,
+	}
+)
+
+func TestRemoteWrite(t *testing.T) {
+	for _, fn := range tests {
+		tc := fn()
+		t.Run(tc.Name, func(t *testing.T) {
+			t.Parallel()
+			runTest(t, tc)
+		})
+	}
+}
+
+func runTest(t *testing.T, tc cases.Test) {
+	ap := cases.Appendable{}
+	writeHandler := remote.NewWriteHandler(logger, &ap)
+	if tc.Writes != nil {
+		writeHandler = tc.Writes(writeHandler)
+	}
+
+	// Start a HTTP server to expose some metrics and a receive remote write requests.
+	m := http.NewServeMux()
+	m.Handle("/metrics", tc.Metrics)
+	m.Handle("/push", writeHandler)
+	s := http.Server{
+		Handler: m,
+	}
+	l, err := net.Listen("tcp", "localhost:")
+	require.NoError(t, err)
+	go s.Serve(l)
+	defer s.Close()
+
+	// Start the collector.
+	app := startCollector(t, l.Addr())
+	defer app.Shutdown()
+
+	time.Sleep(10 * time.Second)
+
+	// Check we got some data.
+	tc.Expected(t, ap.Batches)
+}
+
+// startCollector starts an instance of the collector that scrapes from and sends remote writes to the specified address.
+func startCollector(t *testing.T, addr net.Addr) *service.Collector {
+	scrapeTarget := addr.String()
+	receiveEndpoint := fmt.Sprintf("http://%s/push", addr.String())
+	config := fmt.Sprintf(configFormat, scrapeTarget, receiveEndpoint)
+
+	receivers, err := component.MakeReceiverFactoryMap(prometheusreceiver.NewFactory())
+	require.Nil(t, err)
+	exporters, err := component.MakeExporterFactoryMap(prometheusremotewriteexporter.NewFactory())
+	require.Nil(t, err)
+	processors, err := component.MakeProcessorFactoryMap(batchprocessor.NewFactory())
+	require.Nil(t, err)
+
+	factories := component.Factories{
+		Receivers:  receivers,
+		Exporters:  exporters,
+		Processors: processors,
+	}
+
+	appSettings := service.CollectorSettings{
+		Factories:      factories,
+		ParserProvider: parserprovider.NewInMemory(strings.NewReader(config)),
+		BuildInfo: component.BuildInfo{
+			Command:     "otelcol",
+			Description: "OpenTelemetry Collector",
+			Version:     "tests",
+		},
+		LoggingOptions: []zap.Option{
+			// Turn off the verbose logging from the collector.
+			zap.WrapCore(func(zapcore.Core) zapcore.Core {
+				return zapcore.NewNopCore()
+			}),
+		},
+	}
+	app, err := service.New(appSettings)
+	require.Nil(t, err)
+
+	go func() {
+		if err := app.Run(); err != nil {
+			t.Error(err)
+		}
+	}()
+
```

Comment:
Let's add a shutdown function as well and return it or  make this a global function
```go
	shutdown := func() {
		app.Shutdown()
		// Wait until we are closed.
		<-stateChannel
		for notClosed := true; notClosed; {
			switch state := <-stateChannel; state {
			case service.Closed:
				notClosed = false
			}
		}
	}
```

Reply:
Added your shutdown func as a return value and called it in runTest() instead of directly invoking app.Shutdown()